exploration/selection-declaration.md

```diff
@@ -0,0 +1,296 @@
+(Note that users could do this using a `.local` declaration to split the operand)
+
+We don't support selection on dates in LDML45, but it's easy to conceptualize.
+How can a user write a _selector_ that doesn't mess up a _formatter_?
+
+```
+.input {$d :datetime skeleton=yMMMdjm}
+.match {$d :datetime month=numeric}
+1 {{Today is {$d} in cold cold {$d :datetime month=long}}}
+* {{Today is {$d}}}
+```
+
+## Requirements
+
+_What properties does the solution have to manifest to enable the use-cases above?_
+
+## Constraints
+
+_What prior decisions and existing conditions limit the possible design?_
+
+## Proposed Design
+
+_Describe the proposed solution. Consider syntax, formatting, errors, registry, tooling, interchange._
+
+## Alternatives Considered
+
+_What other solutions are available?_
+_How do they compare against the requirements?_
+_What other properties they have?_
+
+### Allow both local and input declarative selectors with immutability
+
+In this alternative, we modify the syntax to allow selectors to 
+annotate an input variable (as with `.input`)
+or assign a local variable (as with `.local`).
+Either annotation is immutable and results in a Duplicate Declaration error
+if it attempts to annotate a variable previously annotated.
+
+Example:
+```
+.match {$input :function} $local = {$input :function}
+* * {{This annotates {$input} and assigns {$local} a value.}}
+
+.match $local1 = {$input :function} $local2 = {$input :function2}
+* * {{This assigns two locals}}
+
+.input {$input :function}
+.local $local = {$input :function}
+.match {$input :function} {$local :function}
+* * {{This produces two duplicate declaration errors.}}
+```
+
+The ABNF change looks like:
+```abnf
+selector          = expression / declaration
+declaration       = s variable [s] "=" [s] expression
+```
+
+**Pros**
+- Shorthand is consistent with the rest of the syntax
+- Shorthand version works intuitively with minimal input
+- Preserves immutability
+- Produces an error when users inappropriately annotate some items
+
+**Cons**
+- Selectors can't provide additional selection-specific options
+  if the value has already been annotated
+- Doesn't allow multiple selection on the same operand, e.g.
+  ```
+  .input {$date :datetime skeleton=yMdjm}
+  .match {$date :datetime field=month} {$date :datetime field=dayOfWeek}
+  * * {{This message produces a Duplicate Declaration error
+        even though selection is separate from formatting.}}
+  ```
+  However, this design does allow for a local variable to be easily created
+  for the purpose of selection.
+
+#### Allow "Declarative" Selectors with _Mutability_
+
+In this alternative, selectors are treated as declaration-selectors.
+That is, an annotation in a selector works like a `.input`.
+This permits `.match` selectors to be a shorthand when no declarations exist.
+
+It is not an error to overwrite a previous annotation.
+Instead the selector's annotation replaces what came before.
```

Comment:
```suggestion
Instead the selector implicitly shadows the name of its variable operand.
```

(This is another place where the distinction between literal and variable operands is important. You can shadow variables, but changing the value of a literal doesn't make sense.)